### DISCUSSION TITLE
What Inline was pressed inside TextBlock?

### DISCUSSION THREAD

**Discussion by Tulesha** [AvaloniaUI/Avalonia#14527] [Q&A]
Hello.

In our application, we're trying to implement `Hyperlink` control from `WPF`. We need the `Hyperlink` to be displayed inside a `TextBlock` and inherit from `TextElement` (e.g. from `Span` as is done in `WPF`), since we want it to look like rich text.

Since `TextElement` doesn't inherit from `InputElement`, I can't handle `PointerPressed` event inside `Hyperlink`.

So I'm trying to figer out, what `TextElement` pressed inside `OnPointerPressed` method of `TextBlock`. But only I can get is a pressed `TextLine`. Inside a `TextLine` we have `TextRuns`, but they are not related with `Inlines` of `TextBlock`.

``` C#
protected override void OnPointerPressed( PointerPressedEventArgs e )
{
	base.OnPointerPressed( e );

	var clickInfo = e.GetCurrentPoint( this );

	if( clickInfo.Properties.IsLeftButtonPressed )
	{
		var padding = Padding;
		var point = e.GetPosition( this ) - new Point( padding.Left, padding.Top );

		var hit = TextLayout.HitTestPoint( point );
		var charHit = hit.CharacterHit;
		var index = charHit.FirstCharacterIndex + charHit.TrailingLength;
		var lineIndex = TextLayout.GetLineIndexFromCharacterIndex( index, charHit.TrailingLength > 0 );

		var textLine = TextLayout.TextLines[ lineIndex ];
	}
}
```

Can you please say how can I get pressed `TextElement`?

**Comment by jp2masa** [comment]
I don't think that's possible currently, see #10186.

**Comment by Gillibald** [comment]
You need to map inlines to text positions and then use GetTextBounds to hit test against your hyperlink range
